### ISSUE TITLE
Invalid uses of `{{outlet}}` should throw

### ISSUE THREAD

**Issue by chancancode** [emberjs/ember.js#18547]
1. Using `{{outlet}}` in anything other than a route template (e.g. components)
2. Using `{{outlet}}` inside the application template of a non-routable engine (`{{mount}}`)

**Comment by jelhan** [comment]
Should be covered by [ember-template-lint/no-outlet-outside-route](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-outlet-outside-routes.md), isn't it?

**Comment by kategengler** [comment]
I don't think we actually want to do this. We're giving Routing a rethink but also we now have Glimmer Components as Route Templates.

**Comment by NullVoxPopuli** [comment]
additionally, I think people are using outlet outside of route templaets -- but also route templates can be components now, so I'm not sure this issue still makes sense.

will close for now
